corehq/apps/hqmedia/models.py

```diff
@@ -905,7 +919,7 @@ def create_mapping(self, multimedia, form_path, save=True):
                 updated_doc = self.get(self._id)
                 updated_doc.create_mapping(multimedia, form_path)
 
-    def get_media_objects(self, build_profile_id=None, remove_unused=False):
+    def get_media_objects(self, build_profile_id=None, remove_unused=False, multimedia_map=None):
```

Comment:
This function takes 15 seconds for the AWW app on india, which has 16K multimedia references. When a `multimedia_map` is passed in with only 100 references, it drops to less than a second.